solvers/sr/src/essence_solver.rs

```diff
@@ -0,0 +1,24 @@
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use rand::Rng;
+
+pub struct EssenceSolver {
```

Comment:
EssenceSolver isn't a good name for this, I'd say. It's either EssencePrimeSolver or SavileRowAsSolver maybe? But I think the files are at the right place and you will very quickly beat @niklasdewally's minion backend in terms of completeness :)